chore: update changelog (#1401)
holoviz/datashader: CHANGELOG.rst

Version 0.17.0 (2025-01-29)
---------------------------

> This release adds an enhancement that optimizes edge bundling using optimized numba code, breaking changes to the version of Python supported, compatibility changes to several libraries, build updates, bugfixes for PDF scaling, documentation updates, and maintenance work. Thanks to first time contributors `@rhkarls <https://github.com/rhkarls>`_, `@lmcinnes <https://github.com/lmcinnes>`_, `@thomas-reimonn <https://github.com/thomas-reimonn>`_, `@amaloney <https://github.com/amaloney>`_; and regular contributors `@hoxbro <https://github.com/hoxbro>`_ and `@maximlt <https://github.com/maximlt>`_.

Reviewer: ```suggestion
This release adds an enhancement that optimizes edge bundling using optimized numba code. The minimum supported version of Python is now 3.10, along with making pillow and dask optional dependencies. This release also adds compatibility changes to several downstream libraries, build updates, bugfixes, and documentation updates. Thanks to first time contributors `@rhkarls <https://github.com/rhkarls>`_, `@lmcinnes <https://github.com/lmcinnes>`_, `@thomas-reimonn <https://github.com/thomas-reimonn>`_, `@amaloney <https://github.com/amaloney>`_; and regular contributors `@hoxbro <https://github.com/hoxbro>`_ and `@maximlt <https://github.com/maximlt>`_.
```

Breaking Changes:

> - Drop support for Python 3.9 (`#1399 <https://github.com/holoviz/datashader/pull/1399>`_)

Reviewer: ```suggestion
- Drop support for Python 3.9 (`#1399 <https://github.com/holoviz/datashader/pull/1399>`_)
- Making dask and pillow optional dependencies (`#1350 <https://github.com/holoviz/datashader/pull/1350>`_, `#1365 <https://github.com/holoviz/datashader/pull/1365>`_)
```

- Fix documentation builds (`#1391 <https://github.com/holoviz/datashader/pull/1391>`_)

Compatibility:

- Updated environment pins (`#1355 <https://github.com/holoviz/datashader/pull/1355>`_)
- Compatibility with cuDF 24.12 (`#1363 <https://github.com/holoviz/datashader/pull/1363>`_, `#1378 <https://github.com/holoviz/datashader/pull/1378>`_)
- Compatibility with dask 2025.1 and dask-expr 2 (`#1369 <https://github.com/holoviz/datashader/pull/1369>`_, `#1390 <https://github.com/holoviz/datashader/pull/1390>`_, `#1394 <https://github.com/holoviz/datashader/pull/1394>`_)
- Compatibility with pixi v0.40.0 (`#1387 <https://github.com/holoviz/datashader/pull/1387>`_)
- Compatibility with python 3.13 (`#1398 <https://github.com/holoviz/datashader/pull/1398>`_)

Bugfixes:

- PDF scaling bug (`#1382 <https://github.com/holoviz/datashader/pull/1382>`_)

Documentation:

- Update supported Python versions (`#1371 <https://github.com/holoviz/datashader/pull/1371>`_)
- Update links and fix warnings (`#1375 <https://github.com/holoviz/datashader/pull/1375>`_, `#1377 <https://github.com/holoviz/datashader/pull/1377>`_, `#1402 <https://github.com/holoviz/datashader/pull/1402>`)

Maintenance:

> - Remove hard dependencies (`#1350 <https://github.com/holoviz/datashader/pull/1350>`_, `#1365 <https://github.com/holoviz/datashader/pull/1365>`_)

Reviewer: ```suggestion
```

Moved up to breaking changes.

> Version 0.17.0 ()

Reviewer: ```suggestion
Version 0.17.0 (2025-01-29)
```

Just so we don't forget.

---------------------------

> 

Reviewer: There is a bit of text here which say what has happened and who has contributed, I have pasted an example for 0.16.3 below:

``` rst
This release adds an enhancement and compatibility changes with upstream libraries. Thanks to `@raphaelquast <https://github.com/raphaelquast>`_, `@droumis <https://github.com/droumis>`_ and `@hoxbro <https://github.com/hoxbro>`_.
```

Breaking Changes:

- Drop support for Python 3.9.
- ``pillow`` and ``dask`` are no longer required dependencies.

> - Moved to ``pixi`` and ``hatchling``.

Reviewer: This is not a breaking change, afaik.

Author: Ah, the `pixi` and `hatchling` are not user facing, which means it is not a user-facing breaking change.

Enhancements:

- Edge bundling speedup using optimized ``numba`` code by `@lmcinnes <https://github.com/lmcinnes>`_ (`#1383 <https://github.com/holoviz/datashader/pull/1383>`_).

> - Drop Python 3.9 by `@hoxbro <https://github.com/hoxbro>`_ (`#1399 <https://github.com/holoviz/datashader/pull/1399>`_).

Reviewer: Let us remove this as it wasn't an enhancement. I wrongly classified it as an enhancement.

Build:

- Updates to build infrastructure by `@hoxbro <https://github.com/hoxbro>`_ (`#1372 <https://github.com/holoviz/datashader/pull/1372>`_).

> - Fix documentation builds by `@hoxbro <https://github.com/hoxbro>`_ (`#1391 <https://github.com/holoviz/datashader/pull/1391>`_).

Reviewer: It does still not match the format of the other changelogs. 

```suggestion
- Fix documentation builds (`#1391 <https://github.com/holoviz/datashader/pull/1391>`_)
```

Author: It seems like there is a specification for writing these. Do you have it written somewhere that any one can reference in the future?

Bugfixes:

- PDF scaling bug by `@thomas-reimonn <https://github.com/thomas-reimonn>`_ in (`#1382 <https://github.com/holoviz/datashader/pull/1382>`_).

Documentation:

- Updated supported Python versions by `@rhkarls <https://github.com/rhkarls>`_ in the ``Getting Started`` page (`#1371 <https://github.com/holoviz/datashader/pull/1371>`_).
- Updated original paper link by `@amaloney <https://github.com/amaloney>`_ in (`#1375 <https://github.com/holoviz/datashader/pull/1375>`_).
- Redirect ``Topics`` page to `examples.holoviz.org <https://examples.holoviz.org>`_ by `@maximlt <https://github.com/maximlt>`_ (`#1377 <https://github.com/holoviz/datashader/pull/1377>`_).

Maintenance:

- Refactors by `@hoxbro <https:github.com/hoxbro>`_
  - `#1350 <https://github.com/holoviz/datashader/pull/1350>`_
  - `#1354 <https://github.com/holoviz/datashader/pull/1354>`_
  - `#1357 <https://github.com/holoviz/datashader/pull/1357>`_
  - `#1365 <https://github.com/holoviz/datashader/pull/1365>`_
  - `#1367 <https://github.com/holoviz/datashader/pull/1367>`_
  - `#1368 <https://github.com/holoviz/datashader/pull/1368>`_

> Version 0.16.3 (2024-07-04)

Reviewer: See how the bullet points look below and compare them to the ones above. 

Some of the differences
- The username is not there
- pr wrapped in parentheses

(I am writing this from the iPad, so sorry about the short sentence)

Author: Ah, yes I will wrap the PR links in parentheses. Note that all the items in that list are a sublist that states all were done by @hoxbro, but I will move it such that each line is attributed to the author.
